includes/express-checkout/class-wc-payments-express-checkout-button-helper.php

```diff
@@ -394,6 +394,13 @@ public function should_show_express_checkout_button() {
 
 		// Order total doesn't matter for Pay for Order page. Thus, this page should always display payment buttons.
 		if ( $this->is_pay_for_order_page() ) {
+
+			// ECE doesn't work on the Pay For Order Page.
+			$order = wc_get_order( absint( get_query_var( 'order-pay' ) ) );
```

Comment:
Actually, looks like there was an issue with my local install and might have fixed it on https://github.com/Automattic/woocommerce-payments/issues/9089

I had Card Testing Prevention enabled on my local site and it seems it was causing ECE to fail on the Pay for Order page. 

I'll double check tomorrow.